### PR TITLE
Vulkan lights only

### PR DESCRIPTION
**Pull request by apjagdale**
Added support for vulkan lights. [shadow map remaining]
Lights for both GL and Vulkan are UBO's.
Material for Vulkan now is UBO instead of Push Constants.
Changed few LOGE to LOGI.
This PR does not have screenshot fixes.
Depends on PR #1721 
Tested with gvr-multilight and few light tests.

GearVRF DCO signed off by: Abhijit Jagdale a.jagdale@samsung.com

### CODE REVIEW

**Comment by Caprica666** [comment]
Do all of the lighting tests run on Vulkan? Do they produce the same results as GL?

**Comment by apjagdale** [comment]
I ran Vulkan on S8 plus. Here are the results.
Passed:
-twoSpotLightsIlluminate
-canEnableLight
-directAndPointLightsIlluminate
-directLightHasSpecularReflection
-directLightIlluminatesInColor
-directLightRotatedIlluminates
-spotLightHasSpecularReflection
-pointLightHasSpecularReflection
-pointLightAtCornerIlluminates
-pointLightAtFrontAttenuates
-pointLightIlluminatesInColor
-canDisableLightInRenderData

Failed:
-canDisableLight
-spotLightAtFrontAttenuates
-spotLightAtCornerIlluminates
-spotLightIlluminatesInColor
-spotLightAtFrontIlluminates
-pointLightAtFrontIlluminates

File: GVRf/Framework/framework/src/main/jni/vulkan/vk_framebuffer.h [hunk 1]
```diff
@@ -15,7 +15,6 @@
 #ifndef FRAMEWORK_VK_FRAMEBUFFER_H
 #define FRAMEWORK_VK_FRAMEBUFFER_H
 #include "vulkan/vk_imagebase.h"
-#include "vulkan/vulkan.h"
```
Comment (roshanch) [inline GVRf/Framework/framework/src/main/jni/vulkan/vk_framebuffer.h]: please use vulkan_headers.h

**Comment by liaxim** [comment]
Needs rebase

**Comment by liaxim** [comment]
Please rebase

**Comment by apjagdale** [comment]
I will rebase once #1721 is merged

File: GVRf/Framework/framework/src/main/jni/vulkan/vulkanCore.cpp [hunk 2]
```diff
@@ -593,15 +594,15 @@ void VulkanCore::InitCommandPools(){
                                           &descriptorLayout);
         GVR_VK_CHECK(!ret);
 
-
+/*
         VkPushConstantRange pushConstantRange = {};
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/jni/vulkan/vulkanCore.cpp]: What is the point of keeping commented out block?

File: GVRf/Framework/framework/src/main/jni/vulkan/vulkan_render_data.cpp [hunk 3]
```diff
@@ -32,14 +32,15 @@ void VulkanRenderData::render(Shader* shader, VkCommandBuffer cmdBuffer, int cur
     VkDescriptorSet descriptorSet = getDescriptorSet(curr_pass);
     //bind out descriptor set, which handles our uniforms and samplers
     if (!isDescriptorSetNull(curr_pass)) {
+        /*
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/jni/vulkan/vulkan_render_data.cpp]: Why keep the commented out block?

**Comment by liaxim** [comment]
Cleaned and rebuilt few times, switched back to gcc. Still keeps crashing every time for me.

VulkanCore::InitPipelineForRenderData gets called twice before the crash. In the second call vkCreateGraphicsPipelines always returns VK_ERROR_INITIALIZATION_FAILED in my build. What could be the reason for that error?

**Comment by roshanch** [comment]
crashes at my end. shader compilation failure.

**Comment by liaxim** [comment]
Looks good. Will replace the use of assert with std::terminate later. Planning to merge on Thursday.

**Pull request merged**